src/main/java/fi/okm/jod/yksilo/controller/ehdotus/MahdollisuudetController.java

```diff
@@ -67,11 +67,12 @@ class MahdollisuudetController {
     log.info("Creating TyomahdollisuudetController, endpoint: {}", endpoint);
   }
 
-  private static MahdollisuusTyyppi getTyyppi(boolean tyo, boolean koulutus) {
-    if (tyo) return MahdollisuusTyyppi.TYOMAHDOLLISUUS;
-    if (koulutus) return MahdollisuusTyyppi.KOULUTUSMAHDOLLISUUS;
-    throw new IllegalArgumentException(
-        "Unknown mahdollisuus type. Either tyo or koulutus must be true.");
+  private static MahdollisuusTyyppi getTyyppi(String tyyppi) {
+    try {
+      return MahdollisuusTyyppi.valueOf(tyyppi);
+    } catch (IllegalArgumentException e) {
+      throw new IllegalArgumentException(String.format("Unknown mahdollisuus type. %s", tyyppi), e);
+    }
```

Comment:
Perhaps a bit redundant to wrapping a exception with basically the same exception.